### PR TITLE
tracing: put the wire types on the flat call frames too

### PR DESCRIPTION
**Pull request by AskAlexSharov**
The `trace_*` tracer, given the same treatment as #23766. Three generated marshalers go:

- **`flatCallFrame` needed no field change.** Its directive had no `-field-override`, so the generated `MarshalJSON` copied the struct into a byte-for-byte identical shadow and called `json.Marshal` on it. All it bought was making the type a `json.Marshaler`, which sends the enclosing slice down `marshalerEncoder` → `appendCompact` once per frame.
- **`flatCallResult`**: `Code`/`Output` go `*[]byte` → `*hexutil.Bytes`.
- **`flatCallAction`**: `Balance`/`Value` → `*hexutil.U256`, `Init`/`Input` → `*hexutil.Bytes`.

The `(*uint256.Int)(input.Value)` and `[]byte(input.Input)` conversions #23766 had to add at this boundary go away — both sides carry the same type now.

512 flat frames, best of 6, M4 Max. These do not nest, so it is a constant factor, not a removed quadratic:

| | ns/op | B/op | allocs |
|---|---:|---:|---:|
| go1.25.7, before | 2,726,418 | 2.07 MB | 7,692 |
| **go1.25.7, after** | **538,935** | **864 KB** | 4,610 |
| go1.27, before | 1,250,629 | 3.67 MB | 3,105 |
| **go1.27, after** | **571,519** | **471 KB** | **3** |

Output is unchanged: `TestFlatCallFrameJSON` pins the full frame for CALL, DELEGATECALL, CREATE2, SELFDESTRUCT, nil vs empty bytes and the error path. The same expected strings pass on the gencodec code this PR deletes.

Net −179 lines. #23766 has merged, so this stands alone now.

### CODE REVIEW

**Review by domiwei — CHANGES_REQUESTED** [review]
The representation changes look statically equivalent, and I found no concrete production defect. Nil versus non-nil empty bytes, zero and maximum U256 values, field tags, HTML escaping, and the synchronous flatten-then-marshal ownership path all appear preserved.

However, the central claim of this PR is exact native flatCallTracer output preservation after deleting three generated marshalers, and there is currently no exact-output regression test for that tracer. The callTracer golden fixtures exercise a different tracer, TestOeTracer exercises a separate implementation, and the production flat-tracer test only checks for one error substring.

Please add a focused full-output test covering at least CALL, CREATE or CREATE2, SELFDESTRUCT, nil versus empty byte fields, zero and maximum U256 quantities, omissions, and field ordering. That would directly pin the public wire contract this refactor depends on.

**Review by domiwei — APPROVED** [review]
Re-reviewed at 3ce5ffe1ee. The requested wire-format regression coverage is now present. TestFlatCallFrameJSON pins exact output for CALL, DELEGATECALL, CREATE2, SELFDESTRUCT, nil and empty byte fields, zero and maximum U256 quantities, field omissions and ordering, and the error-without-result path.

The production diff is unchanged from the previously reviewed generation, and the focused native and tracetest suites pass. I found no remaining blocking issues.

File: execution/tracing/tracers/native/prestate.go [hunk 1]
```diff
@@ -39,19 +39,17 @@ import (
...
 
 type state = map[accounts.Address]*account
 
 type account struct {
-	Balance *big.Int `json:"balance,omitempty"`
+	Balance *hexutil.Big `json:"balance,omitempty"`
 	// Code is a pointer so omitempty can omit unchanged code (nil) while
 	// still emitting "0x" when code is cleared (e.g. EIP-7702 deauth).
-	Code     *[]byte                     `json:"code,omitempty"`
+	Code     *hexutil.Bytes              `json:"code,omitempty"`
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply execution/tracing/tracers/native/prestate.go:52]: Resolved by merging main: #23766 has landed, so `callFrame`/`account` and `gen_account_json.go` are no longer part of this diff. What is left is exactly the three flat marshalers the description names — `gen_flatcallframe_json.go`, `gen_flatcallaction_json.go`, `gen_flatcallresult_json.go` — plus `call_flat.go` and the tests. Net is now −179 lines, and the description says so.

File: execution/tracing/tracers/native/prestate_test.go [hunk 2]
```diff
@@ -240,3 +240,96 @@ func TestOnlyNilPointersAreOmitted(t *testing.T) {
 	require.NotContains(t, string(b), `"to"`)
 	require.Contains(t, string(b), `"input":"0x"`, "input carries no omitempty")
 }
+
+// The flat tracer's output is the public parity/OE trace shape. These cases pin
+// it exactly — field names, ordering, omissions and quantity encoding — because
+// the wire types on the frame are all that produce it now.
+func TestFlatCallFrameJSON(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply execution/tracing/tracers/native/prestate_test.go:265]: Good catch, that was a real gap — `TestFlatCallFrameJSON` passes with or without the generated marshalers, so it pinned the output but not the reason for the change. Added `TestFlatCallTypesHaveNoJSONMarshaler`, asserting all three of `flatCallFrame`, `flatCallAction` and `flatCallResult`, matching the nearby `callFrame` and `account` tests.

Checked it actually bites: putting a `MarshalJSON` back on `flatCallResult` fails it with `flatCallResult must not implement json.Marshaler`, and removing it passes again.
